ray_on_golem/reputation/updater.py

```diff
@@ -0,0 +1,77 @@
+from contextlib import contextmanager
+from typing import Iterable, Tuple
+
+import aiohttp
+
+from ray_on_golem.reputation import models as m
+
+REPUTATION_SYSTEM_URI = "https://reputation.dev-test.golem.network/v1/"
+REPUTATION_SYSTEM_PROVIDER_SCORES = "providers/scores"
+
+
+class ReputationUpdaterException(Exception):
+    ...
+
+
+class ReputationUpdater:
+    def __init__(self, network: str = "polygon"):
+        self._network = network
+
+    @property
+    def reputation_uri(self):
+        return f"{REPUTATION_SYSTEM_URI}{REPUTATION_SYSTEM_PROVIDER_SCORES}?network={self._network}"
+
+    @contextmanager
+    def _no_progress_bar(self, iterable: Iterable):
+        yield iterable
+
+    async def update(self, progress_bar=None) -> Tuple[int, int, int, int]:
+        if not progress_bar:
+            progress_bar = self._no_progress_bar
+        cnt_added = 0
+        cnt_updated = 0
+        cnt_ignored = 0
+        cnt_total = 0
+        async with aiohttp.request("get", self.reputation_uri) as response:
```

Comment:
Why do we want to be in this `with` in this entire block of code? You can also create a separated method that will return `response.json()`

```python
        if not progress_bar:
            progress_bar = self._no_progress_bar
        
        data = self._get_data_from_api() # ToDo rename
        
        cnt_added = cnt_updated =  cnt_ignored = cnt_total = 0
        with progress_bar(data.get("providers")) as providers:
                        ...
```